### ISSUE TITLE
Replace PLATFORM with SPEC

### ISSUE THREAD

**Issue by mgaudet** [eclipse/omr#701]
Long term solution to #699 will be to retire PLATFORM in favour of better integration with the OMR SPEC system.

**Comment by rwy0717** [comment]
We should really consider dropping both `SPEC` and `PLATFORM` in favour of configure's built in target/build/host [platform triplets](https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Specifying-Target-Triplets.html).

**Comment by mgaudet** [comment]
I agree -- that was more than I was willing to pitch for here tho :)
